From e6abf5de3ce1d661aa6d3a5396fa82510a99a51d Mon Sep 17 00:00:00 2001
From: Goldwyn Rodrigues <rgoldwyn@suse.com>
Date: Thu, 5 Nov 2015 21:50:19 -0600
Subject: [PATCH 04/24] md-cluster: Allow spare devices to be marked as faulty
Git-commit: 09afd2a8d6ad2c40f3c1ae0b3f83784864cf4c15
Patch-mainline: v4.5-rc1
References: fate#316335

If a spare device was marked faulty, it would not be reflected
in receiving nodes because it would mark it as activated and continue.
Continue the operation, so it may be set as faulty.

Signed-off-by: Goldwyn Rodrigues <rgoldwyn@suse.com>
Signed-off-by: NeilBrown <neilb@suse.com>
Acked-by: Guoqing Jiang <gqjiang@suse.com>
---
 drivers/md/md.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/md/md.c b/drivers/md/md.c
index 809799a..ae0b43e 100644
--- a/drivers/md/md.c
+++ b/drivers/md/md.c
@@ -9070,7 +9070,6 @@ static void check_sb_changes(struct mddev *mddev, struct md_rdev *rdev)
 				ret = remove_and_add_spares(mddev, rdev2);
 				pr_info("Activated spare: %s\n",
 						bdevname(rdev2->bdev,b));
-				continue;
 			}
 			/* device faulty
 			 * We just want to do the minimum to mark the disk
-- 
2.6.2

